### ISSUE TITLE
TupleUnpack outputs not used in the correct places in the trace graph

### ISSUE THREAD

**Issue by alondj** [pytorch/pytorch#32930]
## To Reproduce

```python
import torch
import torch.nn as nn


class ABC(nn.Module):
    def __init__(self):
        super(ABC, self).__init__()
        self.l = ReturnTuple()

    def forward(self, x):
        a, b, c = self.l(x)
        # use different ops to distinguish tuple elements
        # the trace should show that a goes to relu and c to softmax
        return a.relu(), b.sigmoid(), c.softmax(1)


class CBA(nn.Module):
    def __init__(self):
        super(CBA, self).__init__()
        self.l = ReturnTuple()

    def forward(self, x):
        a, b, c = self.l(x)
        # use different ops to distinguish tuple elements
        # the trace should show that c goes to relu and a to softmax
        # but looks exactly the same as above
        return c.relu(), b.sigmoid(), a.softmax(1)


class ReturnTuple(nn.Module):
    def forward(self, x):
        # return a tuple
        return x * 3, x * 4, x * 5


if __name__ == "__main__":
    sample = torch.randn(10, 10)
    traced_ABC = torch.jit.trace(ABC(), sample)
    traced_CBA = torch.jit.trace(CBA(), sample)
    scripted_CAB = torch.jit.script(CAB())

    print("ABC trace")
    print(str(traced_ABC.graph))
    print("CBA trace")
    print(str(traced_CBA.graph))
    print("CBA scripted")
    print(str(scripted_CAB.graph))
```

ABC trace
```python
graph(%self.1 : __torch__.torch.nn.modules.module.___torch_mangle_0.Module,
      %x : Float(10, 10)):
  %18 : __torch__.torch.nn.modules.module.Module = prim::GetAttr[name="l"](%self.1)
  %27 : (Tensor, Tensor, Tensor) = prim::CallMethod[name="forward"](%18, %x)
  %24 : Tensor, %25 : Tensor, %26 : Tensor = prim::TupleUnpack(%27)
  %11 : Float(10, 10) = aten::relu(%24) # tuple_unpack.py:12:0
  %12 : Float(10, 10) = aten::sigmoid(%25) # tuple_unpack.py:12:0
  %13 : int = prim::Constant[value=1]() # tuple_unpack.py:12:0
  %14 : None = prim::Constant()
  %15 : Float(10, 10) = aten::softmax(%26, %13, %14) # tuple_unpack.py:12:0
  %16 : (Float(10, 10), Float(10, 10), Float(10, 10)) = prim::TupleConstruct(%11, %12, %15)
  return (%16)
```

CBA trace unexpected behavior
```python
graph(%self.1 : __torch__.torch.nn.modules.module.___torch_mangle_4.Module,
      %x : Float(10, 10)):
  %18 : __torch__.torch.nn.modules.module.___torch_mangle_3.Module = prim::GetAttr[name="l"](%self.1)
  %27 : (Tensor, Tensor, Tensor) = prim::CallMethod[name="forward"](%18, %x)
  %24 : Tensor, %25 : Tensor, %26 : Tensor = prim::TupleUnpack(%27)
  %11 : Float(10, 10) = aten::relu(%24) # tuple_unpack.py:22:0
  %12 : Float(10, 10) = aten::sigmoid(%25) # tuple_unpack.py:22:0
  %13 : int = prim::Constant[value=1]() # tuple_unpack.py:22:0
  %14 : None = prim::Constant()
  %15 : Float(10, 10) = aten::softmax(%26, %13, %14) # tuple_unpack.py:22:0
  %16 : (Float(10, 10), Float(10, 10), Float(10, 10)) = prim::TupleConstruct(%11, %12, %15)
  return (%16)
```
the relu should use the third output of the tuple unpack and not the first

## Expected behavior

in the CBA trace it should show aten::relu(%26) and aten::softmax(%24, %13, %14)
to indicate the tuple element used.
 as in the scripted version
```python
graph(%self : __torch__.CBA,
      %x.1 : Tensor):
  %14 : None = prim::Constant()
  %13 : int = prim::Constant[value=1]() # tuple_unpack.py:22:48
  %2 : __torch__.ReturnTuple = prim::GetAttr[name="l"](%self)
  %4 : (Tensor, Tensor, Tensor) = prim::CallMethod[name="forward"](%2, %x.1) # tuple_unpack.py:21:18
  %a.1 : Tensor, %b.1 : Tensor, %c.1 : Tensor = prim::TupleUnpack(%4)
  %9 : Tensor = aten::relu(%c.1) # tuple_unpack.py:22:15
  %11 : Tensor = aten::sigmoid(%b.1) # tuple_unpack.py:22:25
  %15 : Tensor = aten::softmax(%a.1, %13, %14) # tuple_unpack.py:22:38
  %16 : (Tensor, Tensor, Tensor) = prim::TupleConstruct(%9, %11, %15)
  return (%16)
```

## Environment

Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.14.0

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti

Nvidia driver version: 435.21
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.4                      243  
[conda] mkl-include               2019.4                      243  
[conda] mkl-service               2.3.0            py37he904b0f_0  
[conda] mkl_fft                   1.0.15           py37ha843d7b_0  
[conda] mkl_random                1.1.0            py37hd6b4f25_0  
[conda] pytorch                   1.4.0           py3.7_cuda10.1.243_cudnn7.6.3_0    pytorch
[conda] torchvision               0.5.0                py37_cu101    pytorch

cc @suo @ezyang @gchanan @zou3519

**Comment by wanchaol** [comment]
@alondj I spent sometime on this issue, the conclusion here is: this is not a bug, and it works well if you feed your input and find that the graph is actually correct. The problem here is actually because we are hiding the subgraphs so you don't know what's happening in the `ReturnTuple` subgraph, so if we inline all the graph together for `CBA`, the graph is semantically correct:

```python
torch._C._jit_set_inline_everything_mode(True)
traced_CBA = torch.jit.trace(CBA(), (sample,), check_trace=False)
print(traced_CBA.graph)
```
the graph is
```
CBA trace
graph(%self.1 : __torch__.CBA,
      %x : Float(10, 10)):
  %28 : Long() = prim::Constant[value={5}](), scope: __module.l # test_script.py:81:0
  %29 : Long() = prim::Constant[value={4}](), scope: __module.l # test_script.py:81:0
  %30 : Long() = prim::Constant[value={3}](), scope: __module.l # test_script.py:81:0
  %a : Float(10, 10) = aten::mul(%x, %30), scope: __module.l # test_script.py:81:0
  %b : Float(10, 10) = aten::mul(%x, %29), scope: __module.l # test_script.py:81:0
  %c : Float(10, 10) = aten::mul(%x, %28), scope: __module.l # test_script.py:81:0
  %11 : Float(10, 10) = aten::relu(%c) # test_script.py:75:0
  %12 : Float(10, 10) = aten::sigmoid(%b) # test_script.py:75:0
  %13 : int = prim::Constant[value=1]() # test_script.py:75:0
  %14 : None = prim::Constant()
  %15 : Float(10, 10) = aten::softmax(%a, %13, %14) # test_script.py:75:0
  %16 : (Float(10, 10), Float(10, 10), Float(10, 10)) = prim::TupleConstruct(%11, %12, %15)
  return (%16)
```
Because actually in tracing we generate the subgraph for `ReturnTuple` like this:
```
graph(%self : __torch__.ReturnTuple,
      %x : Float(10, 10)):
  %1 : Long() = prim::Constant[value={3}](), scope: __module.l # test_script.py:81:0
  %a : Float(10, 10) = aten::mul(%x, %1), scope: __module.l # test_script.py:81:0
  %4 : Long() = prim::Constant[value={4}](), scope: __module.l # test_script.py:81:0
  %b : Float(10, 10) = aten::mul(%x, %4), scope: __module.l # test_script.py:81:0
  %6 : Long() = prim::Constant[value={5}](), scope: __module.l # test_script.py:81:0
  %c : Float(10, 10) = aten::mul(%x, %6), scope: __module.l # test_script.py:81:0
  %8 : (Float(10, 10), Float(10, 10), Float(10, 10)) = prim::TupleConstruct(%c, %b, %a)
  return (%8)
```
So this is correct behavior in tracing and it's just that the graph you see is a bit confusing. 

It would be nicer if we preserved the order when doing the `TracedModuleCall` though, I will see if there's a way to do that :)

**Comment by wanchaol** [comment]
seems like there's so easy way to preserve the ordering when fix up the trace scope blocks. I am going to close this as it's not a bug. @alondj feel free to reopen it if you like to discuss more.
